File: 2379_minimum_recolors_to_get_k_consecutive_black_blocks/minimumRecolors.ts
function minimumRecolors(blocks: string, k: number): number {
  const n = blocks.length;
  let m = n;

  for (let i = 0; i < n - k + 1; i++) {
    let curr_m = 0;
    for (let j = i; j < k + i; j++) {
      if (blocks[j] === "W") curr_m++;
    }
    m = Math.min(m, curr_m);
  }

  return m;
}

File: 2379_minimum_recolors_to_get_k_consecutive_black_blocks/minimumRecolors.java
class Solution {
    public int minimumRecolors(String blocks, int k) {
        int n = blocks.length();

        int black = 0;
        int res = n;

        for (int i = 0; i < n; i++) {
            if (i - k >= 0 && blocks.charAt(i - k) == 'B') black--;
            if (blocks.charAt(i) == 'B') black++;
            res = Math.min(res, k - black);
        }

        return res;
    }
}